### ISSUE TITLE
Components on pages disappear in production

### ISSUE THREAD

**Issue by rabbitfufu** [nuxt/nuxt.js#9198]
Hello!
I am using Nuxt to generate a static website.

In layouts/default.vue
```
<template>
<div>
	<SiteHeader />
	<Nuxt />
	<SiteFooter />
</div>
</template>
```
In pages/home.vue
```
<template>
<div>
	<ComponentA />
	<ComponentB />
	<ComponentC />
</div>
</template>

<script>
export default {
	name: 'Home',
	layout: 'default'
}
</script>
```
This renders perfectly in development.

However when I export the site to production (static) and view it, I only see the header and footer.  The three components on the home page flash briefly and then disappear again.

Strangely, if I make the following adjustments...

In layouts/default.vue
```
<template>
<div>
	<SiteHeader />
	<ComponentA />
	<ComponentB />
	<ComponentC />
	<SiteFooter />
</div>
</template>
```
In pages/home.vue
```
<template>
<div />
</template>

<script>
export default {
	name: 'Home',
	layout: 'default'
}
</script>
```
...and then export the site to production, everything works perfectly.

So I assume there is some kind of issue with with how things are being injected through the Nuxt component?  Not sure if there is a setting somewhere I am missing?

Thanks (in advance) for your help!

**Comment by rabbitfufu** [comment]
Thanks for the quick reply!

If I understand correctly,  In pages/home.vue:

```
<template>
<div>
	Testing
</div>
</template>

<script>
export default {
	name: 'Home',
	layout: 'default',
	fetchOnServer: false,
	fetchKey: 'home'
}
</script>
```

Unfortunately this does not work.  "Testing" briefly flashes and then disappears

**Comment by rabbitfufu** [comment]
I tried adding fetchKey to all of the components.  (SiteHeader, ComponentA, ComponentB, ComponentC, SiteFooter) -- unfortunately it does not seem to make a difference.  The problem persists.

**Comment by rabbitfufu** [comment]
@danielroe sure thing

Here is a repo that demonstrates the issue:  https://github.com/rabbitfufu/website

Here is the compiled site (nuxt generate) that shows the problem in action:  https://dev.simpleset.net/home/index.html

Appreciate the help!

**Comment by rabbitfufu** [comment]
@danielroe 

goodness, how embarrassing!  After a bunch of searching, I found there was a piece of invalid markup in my html, causing the server and the client side renders to be different.  (I had an a tag nested inside of another a tag.)
I fixed the markup, and everything works perfectly now.
Apologies for the hassle!  And thanks again for your help, I am really enjoying using this framework for developing static websites.
